cowprotocol/competitor_analysis/cow_alternatives_analysis_4810844.sql

```diff
@@ -0,0 +1,77 @@
+with all_cow_users as (
+    select
+        tx_from as address,
+        array_agg(distinct blockchain) as uses_on_chains
+    from
+        dex_aggregator.trades
+    where
+        project = 'cow_protocol'
+        and
+        block_time between timestamp '{{start_time}}' and timestamp '{{end_time}}'
+    group by
+        1
+),
+
+cow_users_filtered as (
+    select *
+    from
+        all_cow_users
+    where
+        not contains(uses_on_chains, '{{chain_of_interest}}')
+),
+
+chains_supported_by_cow as (
+    select distinct blockchain
+    from
+        dex_aggregator.trades
+    where
+        project = 'cow_protocol'
+),
+
+all_competitor_transactions as (
+    select
+        tx_from as address,
+        blockchain as chain_used_for_competitor,
+        project as competitor_project,
+        sum(amount_usd) as competitor_total_volume_usd,
+        count(*) as competitor_total_transactions
+    from
+        dex.trades
```

Comment:
nit: perform union (with relevant) here to reduce the amount of code duplication in this cte.